mcserver/views.py

```diff
@@ -1310,19 +1886,26 @@ def post(self, request, format='json'):
 @renderer_classes((TemplateHTMLRenderer, JSONRenderer))
 @csrf_exempt
 def verify(request):
-
-    device = request.user.emaildevice_set.all()[0]
-    data = json.loads(request.body.decode('utf-8'))
-    verified = device.verify_token(data["otp_token"])
-    print("VERIFICATION", verified)
-    request.user.otp_verified = verified
-    if 'remember_device' in data and data['remember_device']:
-        request.user.otp_skip_till = timezone.now() + timedelta(days=90)
-    request.user.save()
+    try:
+        device = request.user.emaildevice_set.all()[0]
+        data = json.loads(request.body.decode('utf-8'))
+        verified = device.verify_token(data["otp_token"])
+        print("VERIFICATION", verified)
+        request.user.otp_verified = verified
+
+        if 'remember_device' in data and data['remember_device']:
+            request.user.otp_skip_till = timezone.now() + timedelta(days=90)
+        request.user.save()
+
+    except Exception:
+        if settings.DEBUG:
+            raise Exception(_("error") % {"error_message": str(traceback.format_exc())})
+        raise APIException(_('verification_error'))
 
     if not verified:
-        return Response({
-        }, status.HTTP_401_UNAUTHORIZED)
```

Comment:
we deleted this 401 return. is that correct?

Reply:
I think so, I am raising a NonAuthenticated error instead, which should result in a status 401 Unauthorized as well.